### PR TITLE
Remove existing Self-paced PL course blocks from teacher homepage

### PR DESCRIPTION
**Pull request by kelbyhawn**
Removes the My Professional Learning Courses RecentCourses component on the Teacher Homepage (https://studio.code.org/home), and adds a note and button to view PL courses on the updated Professional Learning page (https://studio.code.org/my-professional-learning#self-paced-pl).
- I added a skip link to the relevant section on the Professional Learning page that was added in this PR: https://github.com/code-dot-org/code-dot-org/pull/57869
- Removed irrelevant unit tests and added a new one to check that this section shows up when a teacher has PL courses
  - I left this as an Enzyme test instead of updating it to React Testing Library since this will be a temporary addition

**Jira ticket:** [ACQ-1716](https://codedotorg.atlassian.net/browse/ACQ-1716) • Parent ticket: [ACQ-1427](https://codedotorg.atlassian.net/browse/ACQ-1427)

----

## Before
Before

## After
Screenshot 2024-04-11 at 12 29 12 PM

### CODE REVIEW

File: apps/src/templates/studioHomepages/TeacherHomepage.jsx [hunk 1]
```diff
@@ -281,14 +284,27 @@ export const UnconnectedTeacherHomepage = ({
...
         {(plCourses?.length > 0 || topPlCourse) && (
-          <RecentCourses
-            courses={plCourses}
-            topCourse={topPlCourse}
-            showAllCoursesLink={true}
-            isProfessionalLearningCourse={true}
-            hasFeedback={hasFeedback}
-          />
+          <section id={'pl-courses-placeholder'} style={{marginBlock: '6rem'}}>
+            <Heading2 visualAppearance="heading-md">
```
Comment (kelbyhawn) [inline apps/src/templates/studioHomepages/TeacherHomepage.jsx:291]: Made this look like an `h3` to match the other headings on this page.

**Review by TurnerRiley — APPROVED** [review]
Looks great!

File: apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx [hunk 2]
```diff
@@ -159,18 +159,18 @@ export default function LandingPage({
...
         {plCoursesStarted?.length >= 1 && (
-          <section>
+          <section id={'self-paced-pl'}>
             <Heading2>{i18n.plLandingSelfPacedProgressHeading()}</Heading2>
             {RenderSelfPacedProgressTable()}
           </section>
         )}
         {deeperLearningCourseData?.length >= 1 && (
-          <div>
+          <section>
```
Comment (kelbyhawn) [inline apps/src/code-studio/pd/professional_learning_landing/LandingPage.jsx:168]: Updated this to add spacing between the Self-Paced PL section and the Online PL Courses section:

| Before | After |
| ---- | ---- |
| Before | After |

File: apps/src/templates/studioHomepages/TeacherHomepage.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/templates/studioHomepages/TeacherHomepage.jsx:291]: Wondering if we should use the `ContentContainer` component like what's being used in `TeacherSections.jsx` to match the other titles on the page?

**Review by cearachew — COMMENTED** [review]
Looks good, thanks! :shipit:

**Review by cearachew — APPROVED** [review]

**Pull request merged**